common/scala/src/whisk/core/database/DocumentFactory.scala

```diff
@@ -176,7 +169,7 @@ trait DocumentFactory[R, W] extends (R => Try[W]) with InMemoryCache[R, W] {
     }
 
     /**
-     * Fetches a raw record of type R from the datastore by its id (and revision if given)
+     * FIXME UPDATE Fetches a raw record of type R from the datastore by its id (and revision if given)
```

Comment:
Turns out making classes immutable is an orthogonal question (although now it's at least possible). I can still do it as part of this PR.